Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,759 @@
+        }
+    });
+
+    GroundPolylinePrimitive._initialized = false;
+    GroundPolylinePrimitive._initPromise = undefined;
+
+    /**
+     * Initializes the minimum and maximum terrain heights. This only needs to be called if you are creating the
+     * GroundPolylinePrimitive synchronously.
+     *
+     * @returns {Promise} A promise that will resolve once the terrain heights have been loaded.
+     *
+     */
+    GroundPolylinePrimitive.initializeTerrainHeights = function() {
+        var initPromise = GroundPolylinePrimitive._initPromise;
+        if (defined(initPromise)) {
+            return initPromise;
+        }
+
+        GroundPolylinePrimitive._initPromise = ApproximateTerrainHeights.initialize()
+            .then(function() {
+                GroundPolylinePrimitive._initialized = true;
+            });
+
+        return GroundPolylinePrimitive._initPromise;
+    };
+
+    // For use with web workers.
+    GroundPolylinePrimitive._initializeTerrainHeightsWorker = function() {
+        var initPromise = GroundPolylinePrimitive._initPromise;
+        if (defined(initPromise)) {
+            return initPromise;
+        }
+
+        GroundPolylinePrimitive._initPromise = ApproximateTerrainHeights.initialize('../Assets/approximateTerrainHeights.json')
+            .then(function() {
+                GroundPolylinePrimitive._initialized = true;
+            });
+
+        return GroundPolylinePrimitive._initPromise;
+    };
+
+    function createShaderProgram(groundPolylinePrimitive, frameState, appearance) {
+        var context = frameState.context;
+        var primitive = groundPolylinePrimitive._primitive;
+        var attributeLocations = primitive._attributeLocations;
+
+        var vs = primitive._batchTable.getVertexShaderCallback()(PolylineShadowVolumeVS);
+        vs = Primitive._appendShowToShader(primitive, vs);
+        vs = Primitive._appendDistanceDisplayConditionToShader(primitive, vs);
+        vs = Primitive._modifyShaderPosition(groundPolylinePrimitive, vs, frameState.scene3DOnly);
+
+        var vsMorph = primitive._batchTable.getVertexShaderCallback()(PolylineShadowVolumeMorphVS);
+        vsMorph = Primitive._appendShowToShader(primitive, vsMorph);
+        vsMorph = Primitive._appendDistanceDisplayConditionToShader(primitive, vsMorph);
+        vsMorph = Primitive._modifyShaderPosition(groundPolylinePrimitive, vsMorph, frameState.scene3DOnly);
+
+        // Access pick color from fragment shader.
+        // Helps with varying budget.
+        var fs = primitive._batchTable.getVertexShaderCallback()(PolylineShadowVolumeFS);
+
+        // Tesselation on these volumes tends to be low,
+        // which causes problems when interpolating log depth from vertices.
+        // So force computing and writing log depth in the fragment shader.
+        // Re-enable at far distances to avoid z-fighting.
+        var colorDefine = appearance instanceof PolylineColorAppearance ? 'PER_INSTANCE_COLOR' : '';
```

Comment:
Hacky...will pretend I did not see this unless there is an easy way to decouple this. 😄